### PR TITLE
Add instuctions for updating cloudflared binary.

### PR DESCRIPTION
**Pull request by iUnknwn**
Adds a section to the DNS-Over-HTTPS guide with instructions for how to ensure the cloudflared daemon is kept up-to-date.

This adds update instructions for cloudflared, which handles one part of issue #175

### CODE REVIEW

**Comment by dschaper** [comment]
Thanks for the submission. I'm not a fan of automated updates without intervention so I'm not going to approve but the other team members may have different opinions.

**Comment by dschaper** [comment]
Screenshot_2020-01-15 Command-line Arguments - Argo Tunnel(1)

**Comment by iUnknwn** [comment]
The docs I added are simply calling that, actually (and then calling systemctl restart) - if you'd prefer, I can alter the docs to simply explain how to do an updating manually, and then just have a note you can automate it by placing it inside the weekly cron folder?

**Comment by dschaper** [comment]
Yeah, that sounds like a better way to document it. Have the manual process be the main topic and then add a note for how to automate. Bonus if you want to do `cron` and `systemd`s timers!

**Comment by iUnknwn** [comment]
@dschaper Done - I've updated the docs so the manual method is described first, and then there's a note on how to do automatic updating via cron.

Also, I made a small comment earlier in the guide that cloudflared will also work with other DoH providers (so the same setup can be used for google DNS for example).

**Comment by dschaper** [comment]
@iUnknwn Can you rebase this on master? There have been some changes.

**Comment by iUnknwn** [comment]
@dschaper Done - rebased onto master.

**Comment by dschaper** [comment]
Thanks, some linting comments, I'll check them over in a bit.

```text
 docs/guides/dns-over-https.md:65:99 MD034/no-bare-urls Bare URL used [Context: "https://8.8.8.8/dns-query"]
docs/guides/dns-over-https.md:156:91 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
docs/guides/dns-over-https.md:156:91 MD047/single-trailing-newline Files should end with a single newline character
```

**Comment by XhmikosR** [comment]
I don't think this patch is correct. If one has installed cloudflared following the existent guide, then cloudflared won't run as root so updating it with its built-in command can mess things up.

I suggest that we focus on finishing #186 first and build on top of that.

**Comment by XhmikosR** [comment]
@iUnknwn can you fetch the latest master and check what still applies? I added some info in #186

**Comment by iUnknwn** [comment]
This overlaps with the majority of my PR - the only thing that still might be useful is the note about how cloudflared can be configured to use other DoH providers. 

Do you want me to update this PR, or open a new one just for that specific improvement?

**Comment by XhmikosR** [comment]
I think you should try keeping most of your changes and we see the diff. There are some differences including the cron job (which I thought is not needed)

**Comment by iUnknwn** [comment]
@XhmikosR OK - all of those changes should be in this PR - do you want me to do a rebase onto the current docs, or just leave what I currently have?

**Comment by XhmikosR** [comment]
Just rebase this branch and add any more info and we'll review it.

**Comment by iUnknwn** [comment]
Done - just rebased on top of master.

File: docs/guides/dns-over-https.md [hunk 1]
```diff
@@ -60,10 +60,12 @@ sudo useradd -s /usr/sbin/nologin -r -M cloudflared
 Proceed to create a configuration file for `cloudflared` by copying the following in to `/etc/default/cloudflared`. This file contains the command-line options that get passed to cloudflared on startup:
 
 ```bash
-# Commandline args for cloudflared
+# Commandline args for cloudflared, using Cloudflare DNS
 CLOUDFLARED_OPTS=--port 5053 --upstream https://1.1.1.1/dns-query --upstream https://1.0.0.1/dns-query
 ```
 
+**Note:** The `cloudflared` binary will work with other DoH providers (for example, you could use https://8.8.8.8/dns-query for Google DNS).
```
Comment (XhmikosR) [inline docs/guides/dns-over-https.md:67]:
```suggestion
**Note:** The `cloudflared` binary will work with other DoH providers (for example, you could use `https://8.8.8.8/dns-query` for Google DNS).
```

File: docs/guides/dns-over-https.md [hunk 2]
```diff
@@ -223,3 +225,23 @@ sudo systemctl daemon-reload
...
+
+If you want to have the system update `cloudflared` automatically, simply place the update commands in the
+file `/etc/cron.weekly/cloudflared-updater.sh`, and adjust permissions:
+
+```bash
+sudo chmod +x /etc/cron.weekly/cloudflared-updater.sh
+sudo chown root:root /etc/cron.weekly/cloudflared-updater.sh
+```
+
+The system will now attempt to update the cloudflared binary automatically, once per week.
```
Comment (XhmikosR) [inline docs/guides/dns-over-https.md:247]:
```suggestion
The system will now attempt to update the cloudflared binary automatically, once per week.
```

**Comment by iUnknwn** [comment]
@XhmikosR done - changes made.

**Comment by iUnknwn** [comment]
...Actually, I'm re-reading this and I notice we now have two updating cloudflared sections, should the section section be renamed "Keeping Cloudflared Up-To-Date" or something like that?

**Comment by XhmikosR** [comment]
You need to merge your update changes with the current one.

**Comment by iUnknwn** [comment]
OK - changes merged. 

That said, I'm wondering if there might be a way to simply this guide considerably. The main reason to keep the manual configuration route is to give people an option to avoid running cloudflared as root, correct?

If so, wouldn't it make more sense to simply write an extension unit file ([as described here](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/system_administrators_guide/sect-Managing_Services_with_systemd-Unit_Files#brid-Managing_Services_with_systemd-Extending_Unit_Config)). That would let people configure the binary to run as its own non-root user, and it would prevent future updates from overwriting the change. 

If we did this, we'd be able to remove all the sections that refer to manual configuration, and I think it would make the guide considerably more readable (we'd no longer need two sub sections for each install, update, and delete step).

**Review by DL6ER — APPROVED** [review]

**Pull request merged**